subprojects/core/src/test/groovy/org/gradle/api/internal/changedetection/state/ZipHasherTest.groovy

```diff
@@ -73,6 +82,254 @@ class ZipHasherTest extends Specification {
         hash1 != hash2
     }
 
+    def "changing manifest attributes changes the hashcode"() {
+        given:
+        def jarfile = tmpDir.file("test.jar")
+        def manifest = new Manifest()
+        def attributes = manifest.getMainAttributes()
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0")
```

Comment:
Required, because `manifest.write` is called by the `JarOutputStream`:

https://docs.oracle.com/javase/7/docs/api/java/util/jar/Manifest.html#write(java.io.OutputStream)

> Writes the Manifest to the specified OutputStream. Attributes.Name.MANIFEST_VERSION must be set in MainAttributes prior to invoking this method.

Reply:
👍